lib/vector2d.rb

```diff
@@ -1,5 +1,19 @@
 # encoding: utf-8
 
+require 'contracts'
+
+class Vector2d
+  include Contracts
+  VectorLike = Or[
+    [Num, Num],
+    {x: Num, y: Num},
+    {'x' => Num, 'y' => Num},
```

Comment:
Space inside { missing.<br>Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.